core/store/migrations/0051_webhook_specs_external_initiators_join.go

```diff
@@ -0,0 +1,40 @@
+package migrations
+
+import (
+	"gorm.io/gorm"
+)
+
+const up51 = `
+ALTER TABLE webhook_specs DROP COLUMN external_initiator_name, DROP COLUMN external_initiator_spec;
+
+CREATE TABLE external_initiator_webhook_specs (
+	external_initiator_id bigint NOT NULL references external_initiators (id) ON DELETE RESTRICT DEFERRABLE INITIALLY IMMEDIATE,
+	webhook_spec_id int NOT NULL references webhook_specs (id) ON DELETE CASCADE DEFERRABLE INITIALLY IMMEDIATE,
+	spec jsonb NOT NULL,
+	PRIMARY KEY (external_initiator_id, webhook_spec_id)
+);
+
+CREATE INDEX idx_external_initiator_webhook_specs_webhook_spec_id ON external_initiator_webhook_specs (webhook_spec_id);
+CREATE UNIQUE INDEX idx_jobs_unique_flux_monitor_spec_id ON jobs (flux_monitor_spec_id);
```

Comment:
these job specific spec IDs are FKs to primary keys I don't think we need this unique index?

Reply:
Yup, we do. Otherwise you could easily have multiple jobs pointing at the same spec.